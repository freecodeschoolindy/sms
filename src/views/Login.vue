<template>
  <div>
    <h1>Login Page</h1>
    <input v-model="email">
    <input v-model="password">
    <button @click="logUserIn">
      Login
    </button>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex'

export default {
  data() {
    return {
      email: 'gwen@example.com',
      password: 'hellomotto'
    }
  },
  computed: {
    ...mapState(['loggedInUser', 'currentUser'])
  },
  methods: {
    ...mapActions(['login']),
    logUserIn() {
      const payload = {
        username: this.email,
        password: this.password
      }
      this.login(payload)
    }
  }
}
</script>
