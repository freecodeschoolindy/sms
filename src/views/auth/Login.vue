<template>
  <div>
    <h1>Login Page</h1>
    <button class="button" @click="logUserIn">
      Github Login
    </button>
  </div>
</template>

<script>
export default {
  data: function () {
    return {};
  },
  methods: {
    logUserIn: function () {
      // Routes to GitHub login page. Once authenticated, GitHub will route back to `/auth/callback?code=1234`
      window.location.href = 'https://github.com/login/oauth/authorize?client_id=b1566423c56583f16eb0';
    }
  }
};
</script>
