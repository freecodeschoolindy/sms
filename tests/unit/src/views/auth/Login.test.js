import Login from '@/views/auth/Login.vue';

import testHelpers from '@@/testHelpers.js';

describe('Login.vue', () => {
  let localVue;
  let store;
  let setup;

  let permissions = ['student'];

  beforeEach(() => {
    setup = testHelpers.setupTestVariables(permissions);
    localVue = setup.localVue;
    store = setup.store;
  });

  test('Renders correctly', async () => {
    const wrapper = await testHelpers.shallowMountWrapper(Login, {
      localVue,
      store
    });

    expect(wrapper)
      .toMatchSnapshot();
  });
});
