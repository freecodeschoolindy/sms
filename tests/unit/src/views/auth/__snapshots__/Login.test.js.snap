// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue Renders correctly 1`] = `
<div>
  <h1>Login Page</h1>
  <label>
    E-mail:
    <input type="email">
  </label>
  <label>
    Password:
    <input type="password">
  </label>
  <button>
    Login
  </button>
</div>
`;
